client/state/data-layer/wpcom/sites/simple-payments/index.js

```diff
@@ -28,33 +28,44 @@ import formatCurrency from 'lib/format-currency';
 import { getFeaturedImageId } from 'lib/posts/utils-ssr-ready';
 
 /**
- * Reduce function for product attributes stored in post metadata
- * @param { Object }  sanitizedProductAttributes object with all sanitized attributes
- * @param { Object } current  item from post_meta to process
- * @returns { Object } sanitizedProductAttributes
+ * Convert custom post metadata array to product attributes
+ * @param { Array } metadata Array of post metadata
+ * @returns { Object } properties extracted from the metadata, to be merged into the product object
  */
-function reduceMetadata( sanitizedProductAttributes, current ) {
-	if ( metaKeyToSchemaKeyMap[ current.key ] ) {
-		sanitizedProductAttributes[ metaKeyToSchemaKeyMap[ current.key ] ] = current.value;
-	}
-	return sanitizedProductAttributes;
+function customPostMetadataToProductAttributes( metadata ) {
+	const productAttributes = {};
+
+	metadata.forEach( ( { key, value } ) => {
+		const schemaKey = metaKeyToSchemaKeyMap[ key ];
+		if ( schemaKey ) {
+			// If the property's type is marked as boolean in the schema,
+			// convert the value from PHP-ish truthy/falsy numbers to a plain boolean.
+			if ( metadataSchema[ schemaKey ].type === 'boolean' ) {
+				value = !! Number( value );
```

Comment:
Is the use of `Number` necessary here? Shouldn't `!! value` suffice? The only difference here I think would be `!! Number( 'true' )` and `!! Number( 'false' )` are always false.